modules/check_in/app/controllers/check_in/v2/appointments_controller.rb

```diff
@@ -36,10 +37,12 @@ def appt_struct_data
         struct.data
       end
 
-      def merge_facilities(appointments)
+      def merge_facilities_and_clinic(appointments)
         appointments.each do |appt|
-          unless appt[:locationId].nil?
+          unless appt[:locationId].nil? && appt[:clinic].nil?
```

Comment:
we should be able to return facility data even if clinic is nil. the nil check for clinic should be inside the conditional